src/test/userRepository.test.ts

```
@@ -0,0 +1,166 @@
+import bcrypt from "bcrypt";
+import mongoose from "mongoose";
+import { ZodError } from "zod";
+import { FIELD_NAMES, ERROR_MESSAGES } from "../constants";
+import { UserModel } from "../models";
+import { userRepository } from "../repositories";
+import { closeConnectionDatabase, connectDatabase } from "../utils";
+
+const MOCK_EMAIL = "mail@mail.com";
+const MOCK_PASSWORD = "password";
+
+function validateDate(date: NativeDate): boolean {
+  const convertedDate = new Date(date);
+  return !isNaN(convertedDate.getTime());
+}
+
+describe("create user", () => {
+  beforeAll(async () => {
+    await connectDatabase();
+  });
+
+  beforeEach(async () => {
+    await UserModel.deleteMany({});
+  });
+
+  afterAll(async () => {
+    await UserModel.deleteMany({});
+    await closeConnectionDatabase();
+  });
+
+  test("should fail: undefined email", async () => {
```

Maybe there is a better to say this than "should fail", as the test should pass. You could say it like "Should throw an error if the email is undefined".